src/audio/volume/volume.c

```diff
@@ -640,7 +640,18 @@ static int volume_ctrl_get_cmd(struct comp_dev *dev,
 				  cdata->chanv[j].channel,
 				  cdata->chanv[j].value);
 		}
-	} else {
+		break;
+	case SOF_CTRL_CMD_SWITCH:
+		comp_dbg(dev, "volume_ctrl_get_cmd(), SOF_CTRL_CMD_SWITCH, cdata->comp_id = %u",
+			 cdata->comp_id);
+		for (j = 0; j < cdata->num_elems; j++) {
+			cdata->chanv[j].channel = j;
+			cdata->chanv[j].value = cd->muted[j];
```

Comment:
Possible logical inversion here?
switch and mute are opposite. The switch value is 1 when not muted.

Reply:
yes, you are right! thanks for spotting this